src/interaction/random_options_explorer.py

```diff
@@ -0,0 +1,36 @@
+"""An explorer that takes random options."""
+
+from typing import Callable, List, Set, Tuple
+
+from gym.spaces import Box
+
+from predicators.src.approaches.random_options_approach import \
+    RandomOptionsApproach
+from predicators.src.interaction import BaseExplorer
+from predicators.src.structs import Action, ParameterizedOption, Predicate, \
+    State, Task, Type
+
+
+class RandomOptionsExplorer(BaseExplorer):
+    """Samples random options."""
+
+    def __init__(self, predicates: Set[Predicate],
+                 options: Set[ParameterizedOption], types: Set[Type],
+                 action_space: Box, train_tasks: List[Task]) -> None:
+        super().__init__(predicates, options, types, action_space, train_tasks)
+        # Set up a random options approach.
+        self._random_options_approach = RandomOptionsApproach(
```

Comment:
okay so this is a third way to do it... so we have:
1) pass the approach.solve function itself into the init
2) instantiate an approach in the explorer class (what you're doing here)
3) copy-paste the logic of approach.solve into the explorer class (what you're doing in the bilevel explorer)

we should pick one

Reply:
let's stick with (2), otherwise there's no guarantee that the bilevel planning explorer would actually be doing bilevel planning

Reply:
oh I remember why I didn't do that. the bilevel planning approach needs the NSRTs. we could add something that `BilevelPlanningApproach` that sets the NSRTs, but that seems bad... thoughts?

Reply:
but doesn't the bilevel planning explorer get the nsrts in its init?

Reply:
yes but `BilevelPlanningApproach` doesn't

Reply:
note that I removed this (2) case here to resolve circular import issues